mrjob/emr.py

```diff
@@ -2558,9 +2561,16 @@ def _usable_clusters(self, exclude=set(), num_steps=1):
         - total compute units for master node
         - time left to an even instance hour
 
-        The most desirable clusters come *last* in the list.
+        :param valid_clusters: A map of cluster id to cluster info with a valid
```

Comment:
Worth noting that this function side-effects valid_clusters, invalid_clusters (it can add new items to these lists).